WebSiteCore/ClientApp/src/components/navmenu/NavMenu.js

```diff
@@ -7,24 +7,26 @@ import { logout } from '../../actions/authActions';
 import DatePicker from 'react-datepicker'
 import './NavMenu.css';
 import 'react-datepicker/dist/react-datepicker.css';
-
+import NumericInput from 'react-numeric-input';
+import 'font-awesome/css/font-awesome.min.css';
 
 class NavMenu extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      date_To : new Date(),
-      date_From : new Date()
+      date_To: new Date(),
+      date_From: new Date()
     }
```

Comment:
Your component connected to the store. Why do we use local state here?  
For optimization, you need to create one global function which should receive two parameters: value and path. 
If you have a simple path : 
```
this.setState({
   `${path}`:value
});
```
But would be better to use immutability helper 
https://github.com/kolodny/immutability-helper